pkg/controller/job/indexed_job_utils.go

```diff
@@ -243,13 +243,18 @@ func parseIndexesFromString(logger klog.Logger, indexesStr string, completions i
 
 // firstPendingIndexes returns `count` indexes less than `completions` that are
 // not covered by `activePods`, `succeededIndexes` or `failedIndexes`.
+// In cases of PodReplacementPolicy as Failed we will include `terminatingPods` in this list.
 func firstPendingIndexes(jobCtx *syncJobCtx, count, completions int) []int {
 	if count == 0 {
 		return nil
 	}
 	active := getIndexes(jobCtx.activePods)
 	result := make([]int, 0, count)
 	nonPending := jobCtx.succeededIndexes.withOrderedIndexes(sets.List(active))
+	if onlyReplaceFailedPods(jobCtx.job) {
+		terminating := getIndexes(controller.FilterTerminatingPods(jobCtx.pods))
+		nonPending = append(nonPending, jobCtx.succeededIndexes.withOrderedIndexes(sets.List(terminating))...)
```

Comment:
```suggestion
		nonPending = nonPending.withOrderedIndexes(sets.List(terminating))
```
The succeeded indexes are already included in `nonPending` in the first line that defines it

Reply:
+1 let's do this one

Reply:
updated.